crates/sui-graphql-rpc/src/types/dynamic_field.rs

```diff
@@ -105,12 +105,12 @@ impl DynamicField {
             // object. Thus, we use the version of the field object to bound the value object at the
             // correct version.
             let obj = MoveObject::query(
-                ctx.data_unchecked(),
+                ctx,
                 self.df_object_id,
-                ObjectLookupKey::LatestAtParentVersion {
-                    version: self.super_.super_.version_impl(),
-                    checkpoint_viewed_at: self.super_.super_.checkpoint_viewed_at,
-                },
+                Object::under_parent(
+                    self.super_.super_.version_impl(),
```

Comment:
cc @emmazzz iirc this impl. is actually incorrect, we need to use the parent object's version, not the wrapper object. The rules for parent child version dont apply to the wrapper object, which is what we're handling here. Not that this pr introduced the bug but would you be able to add a comment here or something